python/cudf/cudf/_lib/csv.pyx

```diff
@@ -0,0 +1,338 @@
+    read_csv as cpp_read_csv,
+    read_csv_args
+)
+from cudf._lib.cpp.io.types cimport (
+    compression_type,
+    quote_style,
+    source_info,
+    table_with_metadata
+)
+from cudf._lib.io.utils cimport make_source_info
+from cudf._lib.move cimport move
+from cudf._lib.nvtx import annotate
+from cudf._lib.table cimport Table
+
+ctypedef int32_t underlying_type_t_compression
+
+
+class Compression(IntEnum):
+    NONE = (
+        <underlying_type_t_compression> compression_type.NONE
+    )
+    INFER = (
+        <underlying_type_t_compression> compression_type.AUTO
+    )
+    SNAPPY = (
+        <underlying_type_t_compression> compression_type.SNAPPY
+    )
+    GZIP = (
+        <underlying_type_t_compression> compression_type.GZIP
+    )
+    BZ2 = (
+        <underlying_type_t_compression> compression_type.BZIP2
+    )
+    BROTLI = (
+        <underlying_type_t_compression> compression_type.BROTLI
+    )
+    ZIP = (
+        <underlying_type_t_compression> compression_type.ZIP
+    )
+    XZ = (
+        <underlying_type_t_compression> compression_type.XZ
+    )
+
+
+cdef read_csv_args make_read_csv_args(
+    filepath_or_buffer,
+    object lineterminator,
+    object quotechar,
+    int quoting,
+    bool doublequote,
+    header,
+    bool mangle_dupe_cols,
+    usecols,
+    object sep,
+    object delimiter,
+    bool delim_whitespace,
+    bool skipinitialspace,
+    names,
+    dtype,
+    int skipfooter,
+    int skiprows,
+    bool dayfirst,
+    compression,
+    thousands,
+    decimal,
+    true_values,
+    false_values,
+    nrows,
+    byte_range,
+    bool skip_blank_lines,
+    parse_dates,
+    comment,
+    na_values,
+    bool keep_default_na,
+    bool na_filter,
+    object prefix,
+    index_col,
+) except +:
+
+    if delim_whitespace:
+        if delimiter is not None:
+            raise ValueError("cannot set both delimiter and delim_whitespace")
+        if sep != ',':
+            raise ValueError("cannot set both sep and delim_whitespace")
+
+    # Alias sep -> delimiter.
+    if delimiter is None:
+        delimiter = sep
+
+    if decimal == delimiter:
+        raise ValueError("decimal cannot be the same as delimiter")
+
+    if thousands == delimiter:
+        raise ValueError("thousands cannot be the same as delimiter")
+
+    if nrows is not None and skipfooter != 0:
+        raise ValueError("cannot use both nrows and skipfooter parameters")
+
+    if byte_range is not None:
+        if skipfooter != 0 or skiprows != 0 or nrows is not None:
+            raise ValueError("""cannot manually limit rows to be read when
+                                using the byte range parameter""")
+
+    cdef source_info c_source_info = make_source_info(filepath_or_buffer)
+    cdef read_csv_args read_csv_args_c = read_csv_args(c_source_info)
+
+    # Reader settings
+    compression = 'none' if compression is None else compression
+    compression = Compression[compression.upper()]
```

Comment:
Can go straight from `None` --> the enum instead of going `None` --> `'none'` --> enum